### PR TITLE
bees should not attack apids

### PR DESCRIPTION
**Pull request by KazooBard**
## About The Pull Request
Bees do not attack apids. I got permission from maintainer PowerfulBacon.

## Why It's Good For The Game
This fixes the bug

Also I am not 100% sure if my code will not pull a coconut jpg from tf2 on the servers and break everything due to two line of code being added but I have high hopes for this one. I deleted one empty line separating things and added two lines of code. I tested the basic stuff and bees don't attack apids and do attack humans etc. 
## Changelog
:cl:
fix: apids are no longer attacked by bees
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
👍

File: code/modules/hydroponics/beekeeping/beebox.dm [hunk 1]
```diff
@@ -18,6 +18,8 @@
 /mob/living/carbon/human/bee_friendly()
 	if(dna?.species?.id == "pod") //bees pollinate plants, duh.
 		return 1
+	if(dna?.species?.id == "apid") //bees don't attack bees, why would they?
```
Comment (Archanial) [inline code/modules/hydroponics/beekeeping/beebox.dm:21]: pretty sure there is a helper for that
`isapid()`

File: code/modules/hydroponics/beekeeping/beebox.dm [hunk 2]
```diff
@@ -18,6 +18,8 @@
...
+		return 1
```
Comment (Archanial) [inline code/modules/hydroponics/beekeeping/beebox.dm]:
```suggestion
		return TRUE
```

File: code/modules/hydroponics/beekeeping/beebox.dm (hunk 1 above)
Comment (KazooBard) [reply code/modules/hydroponics/beekeeping/beebox.dm:21]: I am having problems with the isapid() helper, do you think I can stick with the species id check?

**Comment by PowerfulBacon** [comment]
Cant you just make apids return true on bee friendly though? (Calling bee_friendly on an apid returns true)

**Comment by Sinestia** [comment]
>>This fixes the bug

**Comment by KazooBard** [comment]
> 
> 
> Cant you just make apids return true on bee friendly though? (Calling bee_friendly on an apid returns true)

I did that. I made it return true on apids and false on other species (pods are unattacked)>

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
The bee_friendly proc should return true for felinids. You shouldn't invert the bee_friendly proc for apids, because if they are bee friendly then they become non bee friendly.

**Comment by KazooBard** [comment]
> The bee_friendly proc should return true for felinids. You shouldn't invert the bee_friendly proc for apids, because if they are bee friendly then they become non bee friendly.

What do you mean true for felinids?

**Comment by PowerfulBacon** [comment]
I mean apids.

**Comment by KazooBard** [comment]
> 
> 
> I mean apids.

did it with TRUE instead

**Comment by PowerfulBacon** [comment]
you passed true as a parameter, you need to edit the proc to return true if src is an apid

File: code/modules/hydroponics/beekeeping/beebox.dm (hunk 1 above)
Comment (PowerfulBacon) [reply code/modules/hydroponics/beekeeping/beebox.dm:21]: Way easier would be to give apids TRAIT_BEEFRIEND as an inherent trait

Comment (yorii) [reply code/modules/hydroponics/beekeeping/beebox.dm:21]: I suggest changing line 19 to:
if(isapid(src) || ispodperson(src))

**Review by yorii — CHANGES_REQUESTED** [review]

File: code/modules/hydroponics/beekeeping/beebox.dm (hunk 1 above)
Comment (yorii) [reply code/modules/hydroponics/beekeeping/beebox.dm:21]: try something like this
image

Comment (yorii) [reply code/modules/hydroponics/beekeeping/beebox.dm:21]: Another solution would be to just use the already existing TRAIT_BEEFRIEND and just give that to both apids and podpeople.

**Comment by KazooBard** [comment]
@yorii I will wait for the review of the new changes as they are probably one of the simplest ways to do this. Now I could do it as you asked but I do not know what would it change besides making the line a little shorter. Thanks for the suggestion!

**Review by utilbird — CHANGES_REQUESTED** [review]
just give TRAIT_BEEFRIEND to apids and podperson in `on_species_gain`, then remove the two species checks

(make sure you remove them in the species remove proc too)

**Comment by yorii** [comment]
> just give TRAIT_BEEFRIEND to apids and podperson in `on_species_gain`, then remove the two species checks
> 
> (make sure you remove them in the species remove proc too)

It would seem apids have already had TRAIT_BEEFRIEND for a year now, not sure if this PR is irrelevant or if maybe the original code doesn't work.

**Pull request closed without merging**